### DISCUSSION TITLE
AuthenticationException: Unknown error

### DISCUSSION THREAD

**Discussion by DimX00** [ianjwhite99/sync-connect-sdk#66] [Q&A] [answered]
I am able to get the information that I need but the Traceback shows Unknown error at the end.

```
Traceback (most recent call last):
  File "c:\Scripts\ford.py", line 14, in <module>
    output = vehicle.odometer()
  File "C:\Python39\lib\site-packages\connectedcar\vehicle.py", line 181, in odometer
    json = self.status()
  File "C:\Python39\lib\site-packages\connectedcar\vehicle.py", line 31, in status
    response = self.api.get(
  File "C:\Python39\lib\site-packages\connectedcar\api.py", line 59, in get
    return requester.call('GET', url, headers=headers)
  File "C:\Python39\lib\site-packages\connectedcar\requester.py", line 47, in call
    raise e
  File "C:\Python39\lib\site-packages\connectedcar\requester.py", line 31, in call
    raise E.AuthenticationException(response)
connectedcar.exceptions.AuthenticationException: Unknown error
```

Any idea why this is happening?

**Comment by ianjwhite99** [comment]
I need more information. What request are you trying to make here? I can see it's using the Vehicle class, but what method are you trying to call?

**Reply by DimX00** [reply]
I am just calling vehicle.odometer(). I am able to get the odometer just fine but the traceback shows there was an unexpected error. This is the entire code that I am using from the test file.

```
import connectedcar

client = connectedcar.AuthClient(
    '9fb503e0-715b-47e8-adfd-ad4b7770f73b', None, None, None,)  # Create client connection
access = client.get_user_access_token('cccc', 'vvvvv!')  # Fetch client access token

user = connectedcar.User(access['access_token'])  # New User Object
vehicles = user.vehicles()  # Fetch list of user vehicles

for userVehicle in vehicles:  # For each user vehicle
    vehicle = connectedcar.Vehicle(userVehicle['vin'], access['access_token'])  # Create vehicle object
    output = vehicle.odometer()
    x = output['data']
    print(x['value'])  # Print vehicle odometer value
```

**Reply by ianjwhite99** [reply]
Thanks for that info. I will take a look into this shortly

**Comment by DimX00** [comment, marked as answer]
I was able to figure it out. I am not sure why but the Traceback started showing me the following error in addition to the unknown error. 

`requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://usapi.cv.ford.com/api/vehicles/v5/XXXXXXXXXXXXXXXX/status`

This was helpful because I was able to lookup the vehicle XXXXXXXXXXXXXXXXX in FordPass and noticed it was an older 2014 truck. We have about 15 vehicles in FordPass. I excluded all vehicles under 2019 from my request and now I don't get the error anymore. 

So the above error was because I was requesting information from vehicles that didn't have this data in FordPass because they are not equipped to send this information.

**Reply by ianjwhite99** [reply]
Great, thanks for letting me know
